### DISCUSSION TITLE
Function tensor syntax is confusing

### DISCUSSION THREAD

**Discussion by aborgna-q** [Quantinuum/guppylang#357] [General]
#350 introduces support for calling tuples of functions
```python
c1 = (foo, bar, baz)(a, b.x, c1.x, b, c1, c2)
c2 = (foo, (bar, baz))(a, b.x, c1.x, b, c1, c2)
```

That is cool and all, but I feel most users are not used to that syntax so they'll get confused.

I would rather prefer an explicit `FnTensor(foo, bar, baz)` or similar to clearly lay out what's going on there.

**Comment by mark-koch** [comment]
@croyzor

**Comment by croyzor** [comment]
Personally, I prefer the existing syntax because it's succinct and there's intuition behind it.

However, it leads to the issue of having to suppress python warnings about tuples not being callable (ha!) which im glad to get rid of, so I don't mind changing to something more verbose
